### ISSUE TITLE
Unnessecary Sidebar updates??

### ISSUE THREAD

**Issue by IIHERO4** [andrei1058/SiderbarLib#5]
Hello, i noticed in our server there was alot of cpu-time going to sending scoreboard (in Bedwars1058) so which at time i thought it was normal.

I looked into the source and i saw that the plugin sends alot of unnessecary packets, i fixed it by checking prefix and suffix before and after setting `content` which is already a huge improvement as **for my case** the only thing changing from packet to packet is `Sidebar.suffix` and `Sidebar.prefix`.

Although this change worked nicely for my case, It needs a look from you as developers that uses your library may implement their own `ScoreLine` which will cause unexpected behavior if we are just checking for prefix and suffix.

My Patch:
```java
    public void refreshPlaceholders() {
        for (ScoreLine str : this.lines) {
            if (str.text.isHasPlaceholders()) {
                // Some code here in parsing content
                
                String oldSuffix = str.suffix;
                String oldPrefix = str.prefix;
                str.setContent(content);

                // ignore Intellij Idea warning if present
                if (!oldSuffix.equals(str.suffix) || !oldPrefix.equals(str.prefix)) { 
                    str.sendUpdate();
                }
            }
        }
    }
   ```
   
   This is old version of bw1058, But its equivilant to https://github.com/andrei1058/SiderbarLib/blob/6731c0aca53cf0390b14a08f9e55966bdf1faee1/sidebar-base/src/main/java/com/andrei1058/spigot/sidebar/WrappedSidebar.java#L138-L145

**Comment by IIHERO4** [comment]
Just FYI, When running the plugin with this code and only 1 player is online*

```java
                .....

                if (!oldSuffix.equals(str.suffix) || !oldPrefix.equals(str.prefix)) {
                    System.out.println("UPDATING");
                    str.sendUpdate();
                } else {
                    System.out.println("NOT UPDATING");
                }
```

![image](https://user-images.githubusercontent.com/46909116/156833095-be8b8c13-e612-4704-a2cb-dd7988117db1.png)

`[20:21:14 INFO]: [Server Thread] CraftPlayer{name=KYSHERO}D` are printed between runs of the task

In my case, we are saving 7 packets per player, This is the scoreboard that is being rendered. (Basically only the a packet is sent for the gen time), This also Saved alot of packets when a player is in the main lobby as almost no changes is done while in lobby
![image](https://user-images.githubusercontent.com/46909116/156834029-0fd654ef-1bcf-44d2-ac3f-9ce1143de53b.png)

**Comment by andrei1058** [comment]
I will consider that! Thanks for your time.

**Comment by andrei1058** [comment]
implemented with #15
